cou/steps/hypervisor.py

```diff
@@ -211,18 +256,21 @@ def generate_upgrade_plan(self, target: OpenStackRelease, force: bool) -> Upgrad
         :rtype: UpgradePlan
         """
         plan = UpgradePlan("Upgrading all applications deployed on machines with hypervisor.")
+        for az, group in self.azs.items():
+            hypervisor_plan = HypervisorUpgradePlan(f"Upgrade plan for '{group.name}' to {target}")
 
-        # pre upgrade steps
-        logger.debug("generating pre upgrade steps")
-        plan.add_step(self._generate_pre_upgrade_plan())
+            # pre upgrade steps
+            logger.debug("generating pre upgrade steps")
+            hypervisor_plan.sub_steps.extend(self._generate_pre_upgrade_steps(target, group))
 
-        # upgrade steps
-        for az, group in self.azs.items():
+            # upgrade steps
             logger.debug("generating upgrade steps for %s AZ", az)
```

Comment:
Please keep the debug message consistent across different stages of upgrades

Reply:
What do you mean?

Reply:
You are not logging AZ name in pre-upgrade and post_upgarde steps. It would be better to align them. This was meant to be a nit comment though.

Reply:
Ah now I see, yes because originally it was not part of for loop. I'll change it.